### PR TITLE
#98 Fix/first time logged

### PR DESCRIPTION
**Pull request by deimonbody**

### CODE REVIEW

File: frontend/src/components/main-page/main-page.tsx [hunk 1]
```diff
@@ -9,13 +9,15 @@ const MainPage: React.FC = () => {
...
   useEffect(() => {
-    if (!user?.firstName) {
-      navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/1`);
-      return;
-    }
-    if (!user?.isCompleteTest) {
-      navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/2`);
-      return;
+    if (user) {
+      if (!user?.firstName) {
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/main-page/main-page.tsx]: you already checked that user is not null
```suggestion
      if (!user.firstName) {
```

File: frontend/src/components/main-page/main-page.tsx [hunk 2]
```diff
@@ -9,13 +9,15 @@ const MainPage: React.FC = () => {
...
+        navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/1`);
+        return;
+      }
+      if (!user?.isCompleteTest) {
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/main-page/main-page.tsx]:
```suggestion
      if (!user.isCompleteTest) {
```

File: frontend/src/components/okr/tabs/two-tabs.tsx [hunk 3]
```diff
@@ -14,13 +14,15 @@ function ControlledTabs(): React.ReactElement {
...
-    }
-    if (!user?.isCompleteTest) {
-      navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/2`);
-      return;
+    if (user) {
+      if (!user?.firstName) {
+        navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/1`);
+        return;
+      }
+      if (!user?.isCompleteTest) {
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/okr/tabs/two-tabs.tsx]:
```suggestion
      if (!user.firstName) {
        navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/1`);
        return;
      }
      if (!user.isCompleteTest) {
```

File: frontend/src/components/opportunities-page/opportunities-page.tsx [hunk 4]
```diff
@@ -21,13 +21,15 @@ const OpprotunitiesPage: React.FC = () => {
...
-    }
-    if (!user?.isCompleteTest) {
-      navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/2`);
-      return;
+    if (user) {
+      if (!user?.firstName) {
+        navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/1`);
+        return;
+      }
+      if (!user?.isCompleteTest) {
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/opportunities-page/opportunities-page.tsx]:
```suggestion
      if (!user.firstName) {
        navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/1`);
        return;
      }
      if (!user.isCompleteTest) {
```

File: frontend/src/components/profile-settings/steps/first-step.tsx [hunk 5]
```diff
@@ -60,3 +60,6 @@ const FirstStep: React.FC<IProfileSettingStep> = ({
 };
 
 export default FirstStep;
+// function useAppForm<T>(arg0: { defaultValues: { [x: string]: string; }; validationSchema: import("joi").ObjectSchema<any>; }): { control: any; errors: any; isValid: any; handleSubmit: any; } {
+//   throw new Error('Function not implemented.');
+// }
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/profile-settings/steps/first-step.tsx]: remove comments

File: frontend/src/components/profile/profile.tsx [hunk 6]
```diff
@@ -18,13 +18,15 @@ const ProfileInfo: React.FC = () => {
...
-      return;
+    if (user) {
+      if (!user?.firstName) {
+        navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/1`);
+        return;
+      }
+      if (!user?.isCompleteTest) {
+        navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/2`);
+        return;
+      }
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/profile/profile.tsx]: can we move this repeated logic to separate function and reuse in all files?

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]

**Review by AndriyCh1 — APPROVED** [review]

**Review by nikita-remeslov — COMMENTED** [review]
conflicts

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**